lib/multiplexer/proxyline.go

```diff
@@ -74,24 +98,48 @@ func (p *ProxyLine) Bytes() []byte {
 			SourcePort:      uint16(p.Source.Port),
 			DestinationPort: uint16(p.Destination.Port),
 		}
-		copy(addr4.Source[:], p.Source.IP.To4())
-		copy(addr4.Destination[:], p.Destination.IP.To4())
+		ipv4 := p.Source.IP.To4()
+		if ipv4 == nil {
+			return nil, trace.BadParameter("could not get IPv4 address representation from %q", p.Source.IP.String())
+		}
+		copy(addr4.Source[:], ipv4)
+		ipv4 = p.Destination.IP.To4()
+		if ipv4 == nil {
+			return nil, trace.BadParameter("could not get IPv4 address representation from %q", p.Destination.IP.String())
```

Comment:
I think we could make it clearer whether or not it was the destination or source here - I'd probably not reuse the `ipv4` variable here.